File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    // 심사 시간 정렬
    times.sort((a, b) => a - b);
    let left = 0;
    let right = times[times.length - 1] * n;
    var answer = 0;
    
    while (left <= right) {
        const mid = Math.floor((left + right) / 2);
        let complete = 0;
        for (i=0; i<times.length; i++) {
            complete += Math.floor(mid / times[i]);
        }
        if (complete < n) {
            left = mid + 1;
        } else {
            right = mid - 1;
            answer = mid;
        }
    }
    return answer;
}

File: 프로그래머스/3/43238. 입국심사/입국심사.py
def solution(n, times):
    # 심사 최대/최소 시간
    times.sort()
    left = 0
    right = times[-1] * n
    
    answer = 0
    while left <= right:
        mid = (left+right) // 2
        
        people = 0
        for i in range(len(times)):
            people += mid // times[i]
        
        if people < n:
            left = mid + 1
        else: # 모두 심사 완료
            right = mid - 1
            answer = mid
        
    return answer